kube-runtime/src/controller/mod.rs

```diff
@@ -268,15 +268,16 @@ where
         move |s| {
             Runner::new(scheduler(s), move |request| {
                 let request = request.clone();
+                let reconciler_span = info_span!("reconciling object", "object.ref" = %request.obj_ref, object.reason = tracing::field::Empty);
                 match store.get(&request.obj_ref) {
                     Some(obj) => {
-                        let reconciler_span = info_span!("reconciling object", "object.ref" = %request.obj_ref, object.reason = %request.reason);
+                        reconciler_span.record("object.reason", &tracing::field::display(&request.reason));
```

Comment:
Why not fill in the reason when creating the `span`? We still have access to it there.

Reply:
I'd started out wanting to re-use the span later for the error variant, but actually it's not needed. I'll collapse it again.